tools/slo-generator/slo_generator/backends/elasticsearch.py

```diff
@@ -20,60 +20,60 @@
 
 from elasticsearch import Elasticsearch
 
-from slo_generator.backends.base import MetricBackend
-
 LOGGER = logging.getLogger(__name__)
 
+DEFAULT_DATE_FIELD = '@timestamp'
+
+
+class ElasticsearchBackend:
+    """Backend for querying metrics from ElasticSearch.
 
-class ElasticsearchBackend(MetricBackend):
-    """Backend for querying metrics from ElasticSearch."""
+    Args:
+        client (elasticsearch.ElasticSearch): Existing ES client.
+        es_config (dict): ES client configuration.
+    """
 
-    def __init__(self, **kwargs):
-        self.client = kwargs.get('client')
+    def __init__(self, client=None, **es_config):
+        self.client = client
         if self.client is None:
-            self.client = Elasticsearch(**kwargs)
+            self.client = Elasticsearch(**es_config)
 
-    def good_bad_ratio(self, **kwargs):
+    # pylint: disable=unused-argument
+    def good_bad_ratio(self, timestamp, window, slo_config):
         """Query two timeseries, one containing 'good' events, one containing
         'bad' events.
 
         Args:
             timestamp (int): UNIX timestamp.
             window (int): Window size (in seconds).
-            kwargs (dict): Extra arguments needed by this computation method.
-                project_id (str): GCP project id to fetch metrics from.
-                measurement (dict): Measurement config.
-                    filter_good (str): Query filter for 'good' events.
-                    filter_bad (str): Query filter for 'bad' events.
+            slo_config (dict): SLO configuration.
 
         Returns:
             tuple: A tuple (good_event_count, bad_event_count)
         """
-        window = kwargs['window']
-        index = kwargs['measurement']['index']
-        date = kwargs['measurement'].get('date_field', 'timestamp')
-        query_good = kwargs['measurement']['query_good']
-        query_bad = kwargs['measurement'].get('query_bad')
-        query_valid = kwargs['measurement'].get('query_valid')
+        measurement = slo_config['backend']['measurement']
+        index = measurement['index']
+        query_good = measurement['query_good']
+        query_bad = measurement.get('query_bad')
+        query_valid = measurement.get('query_valid')
+        date_field = measurement.get('date_field', DEFAULT_DATE_FIELD)
```

Comment:
`@timestamp` is the default field in ElasticSearch for the date. In case the user does not pass a `date_field`, it will default correctly to the timestamp ElasticSearch automatically adds on every record. See https://discuss.elastic.co/t/what-is-the-difference-between-timestamp-and-timestamp/46464 for more info